### ISSUE TITLE
Add support for storing data that can be retrieved at runtime (e.g. static variables, key-value pairs, and more)

### ISSUE THREAD

**Issue by anna-geller** [PrefectHQ/prefect#6820]
### First check

- [X] I added a descriptive title to this issue.
- [X] I used the GitHub search to find a similar request and didn't find it.
- [X] I searched the Prefect documentation for this feature.

### Prefect Version

2.x

### Describe the current behavior

Currently, to store variables, users leverage JSON Blocks. However, blocks are most helpful for connecting stateful configuration with business logic and might be too "heavy" for simple use cases that require storing small pieces of data such as key-value pairs, lookup values or even entire JSON objects.

### Describe the proposed behavior

- Add a lightweight interface allowing users to store such data 
- Make this performant enough even if this data needs to be frequently updated and retrieved

### Example Use Cases

- Backfilling
- Storing threshold values
- Simple lookups (e.g. last timestamp scraped)

### Additional context

Related discussion https://github.com/PrefectHQ/prefect/issues/6795

**Comment by anna-geller** [comment]
@EmilRex made a really good point that for secrets, it’s pretty common to keep them in env vars and refer to them using all caps and underscores. This doesn't match our blocks naming convention (all small letters, no underscores, etc) and secrets are not really blocks because they essentially have no capabilities. Their only purpose is storing some static value in an encrypted way and securely retrieving it at runtime.

A different story is when a secret (say AWS credentials) is paired with some capabilities or business logic (e.g. create a boto3 session, connect to boto3 client or resource) - this is a candidate for a real block, but secrets and simple KV pairs are not

## TL;DR
- We need a way to securely store and retrieve simple values such as environment variables
- The naming for this should be not restrictive -- if users want to use all caps, underscores and special characters in their naming, they should be able to do it

**Comment by anna-geller** [comment]
For now the recommended way is to leverage the Secret block, later we'll likely have a dedicated store for that, as discussed in this issue
